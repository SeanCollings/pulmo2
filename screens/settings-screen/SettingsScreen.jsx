import React, { useState, useEffect } from 'react';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import VersionCheck from 'react-native-version-check-expo';
import Constants from 'expo-constants';

import options from './options';
import { useTheme } from '../../hooks/useTheme';
import ThemeSelector from '../../components/ThemeSelector';
import {
  EXPO_VERSION,
  ANDROID_VERSION,
  RELEASE_ID,
} from '../../release-constants';
import CategoryContainer from '../../components/CategoryContainer';
import BarSelector from '../../components/BarSelector';

export const settingsScreenOptions = options;

const getVersion = (version) => (version.includes('#') ? '0' : version);

const SettingsScreen = ({ navigation }) => {
  const theme = useTheme();
  const [latestVersion, setLatestVersion] = useState(null);
  const [showLatestVersion, setShowLatestVersion] = useState(false);

  useEffect(() => {
    let mounted = true;
    VersionCheck.getLatestVersion().then((version) => {
      if (mounted) {
        setLatestVersion(version);
      }
    });
    return () => {
      mounted = false;
    };
  }, []);

  const releaseId = getVersion(RELEASE_ID);
  const androidBuildVersion = getVersion(ANDROID_VERSION);
  const sdkVersion = Constants?.manifest?.sdkVersion.split('.')[0] || '0';

  const opacity = theme.DARK ? 0.36 : 0.7;

  return (
    <View style={{ ...styles.container, backgroundColor: theme.BACKGROUND }}>
      <View style={styles.upperContainer}>
        <CategoryContainer header="Day/night mode">
          <ThemeSelector />
        </CategoryContainer>
        <View style={styles.barSelectorContainer}>
          <BarSelector
            textContent="Disclaimer"
            onPress={() => navigation.navigate('Disclaimer')}
            iconName="note-text"
          />
        </View>
      </View>
      <TouchableOpacity
        onPress={() => setShowLatestVersion((curr) => !curr)}
        activeOpacity={1}
      >
        {showLatestVersion && (
          <View style={styles.versionContainer}>
            <Text
              style={{ ...styles.textStyle, color: theme.TEXT, opacity }}
            >{`latest version: ${latestVersion}`}</Text>
          </View>
        )}
        {!showLatestVersion && (
          <View style={styles.versionContainer}>
            <Text style={{ ...styles.textStyle, color: theme.TEXT, opacity }}>
              version:
            </Text>
            <Text
              style={{
                ...styles.textStyle,
                color: theme.TEXT,
                opacity,
              }}
            >{` ${EXPO_VERSION}.${sdkVersion}.${releaseId}.${androidBuildVersion}`}</Text>
          </View>
        )}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    paddingTop: 10,
  },
  upperContainer: {
    flex: 1,
    width: '100%',
    alignItems: 'center',
  },
  barSelectorContainer: {
    width: '100%',
  },
  versionContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingBottom: 5,
    justifyContent: 'center',
  },
  textStyle: {
    fontFamily: 'tit-light',
    fontSize: 17,
    alignSelf: 'flex-end',
  },
});

export default SettingsScreen;
