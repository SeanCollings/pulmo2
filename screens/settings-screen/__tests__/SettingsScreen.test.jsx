import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import SettingsScreen, { settingsScreenOptions } from '../SettingsScreen';

jest.useFakeTimers();
// jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('SettingsScreen - unit test', () => {
  let tree;
  const navigation = { setOptions: () => {} };

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test('should render', async () => {
    await act(async () => {
      tree = create(<SettingsScreen navigation={navigation} />);
    });

    expect(tree).toMatchSnapshot();
  });

  describe('settingsScreenOptions', () => {
    test('options', () => {
      const options = settingsScreenOptions();
      expect(options).toMatchSnapshot();
    });

    test('headerTitle', () => {
      const headerTitle = settingsScreenOptions().headerTitle({
        tintColor: 'white',
      });
      expect(headerTitle).toMatchSnapshot();
    });
  });
});
