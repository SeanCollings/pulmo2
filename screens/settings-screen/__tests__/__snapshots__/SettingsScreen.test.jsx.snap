// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen - unit test settingsScreenOptions headerTitle 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontSize": 20,
    }
  }
>
  Settings
</Text>
`;

exports[`SettingsScreen - unit test settingsScreenOptions options 1`] = `
Object {
  "headerTitle": [Function],
}
`;

exports[`SettingsScreen - unit test should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "width": "85%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#5bacf3",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 2,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "tit-regular",
              "fontSize": 16,
              "opacity": 1,
              "paddingLeft": 10,
              "textAlignVertical": "center",
            }
          }
        >
          Day/night mode
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3e4b57",
                "fontFamily": "tit-regular",
                "fontSize": 17,
                "opacity": 1,
              }
            }
          >
            Use dark mode
          </Text>
          <View>
            <View
              style={
                Object {
                  "paddingTop": 1,
                }
              }
            >
              <RCTSwitch
                accessibilityRole="button"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onTintColor="#002f56"
                style={
                  Object {
                    "height": 31,
                    "width": 51,
                  }
                }
                testID="customSwitch_valueChange"
                thumbTintColor=""
                tintColor="#CCC"
                value={false}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "elevation": 5,
            "height": 50,
            "justifyContent": "center",
            "marginVertical": 2,
            "opacity": 1,
            "paddingHorizontal": "10%",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2.84,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "justifyContent": "center",
                  "paddingRight": 20,
                  "paddingTop": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#CCC",
                      "fontSize": 21,
                    },
                    undefined,
                    Object {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-regular",
                  "fontSize": 18,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              Disclaimer
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingTop": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#CCC",
                    "fontSize": 23,
                  },
                  undefined,
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 0.7,
        }
      }
    >
      version:
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "flex-end",
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 14,
          "opacity": 0.7,
        }
      }
    >
       1.0.0.1
    </Text>
  </View>
</View>
`;
